azure-iot-device/tests/provisioning/models/test_registration_result.py

```diff
@@ -53,27 +44,53 @@ def test_registration_result_instantiated_correctly(self):
         assert registration_result.registration_state.last_update_date_time == fake_last_update_dttm
         assert registration_result.registration_state.etag == fake_etag
 
+    @pytest.mark.it("Has a to string representation composed of registration state and status")
+    def test_registration_result_to_string(self):
+        fake_registration_state = create_registraion_state()
+        registration_result = create_registration_result(fake_registration_state)
+
+        string_repr = "\n".join([str(fake_registration_state), fake_status])
+        assert str(registration_result) == string_repr
+
     @pytest.mark.it("Has attributes that do not have setter")
     def test_some_properties_of_result_are_not_settable(self):
-        registration_result = RegistrationResult("RequestId123", "Operation456", "emitted", None)
+        registration_result = create_registration_result()
         with pytest.raises(AttributeError, match="can't set attribute"):
-            registration_result.request_id = "MyNimbus2000"
+            registration_result.request_id = "RequestId123"
             registration_result.operation_id = "WhompingWillow"
-            registration_result.status = "Flying"
+            registration_result.status = "Apparating"
             registration_result.registration_state = "FakeRegistrationState"
 
     @pytest.mark.it("Has `RegistrationState` with properties that do not have setter")
```

Comment:
>Has `RegistrationState` with properties that do not have setter" [](start = 21, length = 64)

As above, parametrize this. As is, this does not actually test all properties. It only tests the first one that fails (device id)